### PR TITLE
New Entities and APIs related to CHO mobile application

### PR DESCRIPTION
**Pull request by shreypatidar-beehyv**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.  

# Type of change  

- [ ] Bug fix  
- [x] New feature  
- [ ] Enhancement  
- [ ] Refactoring  
- [ ] Documentation  
- [ ] Other ( please specify )  

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by drtechie** [comment]
@Shrey475 please fix the code vulnerability warnings.

**Comment by Shrey475** [comment]
@drtechie I tried to keep the controller method signatures similar to the existing ones i.e. why I kept it like this. I have changed it now.

File: src/main/environment/common_ci.properties [hunk 1]
```diff
@@ -12,7 +12,9 @@ carestreamOrderCreateURL =@env.COMMON_API_BASE_URL@/carestream/createOrder
 #  Registration
 registrationUrl =@env.COMMON_API_BASE_URL@/beneficiary/create
 
-syncSearchByLocation =@env.COMMON_API_BASE_URL@/id/searchByDistrictId
+syncSearchByLocation =@env.COMMON_API_BASE_URL@/id/searchByVillageIdAndLastModifiedDate
+
+getBenCountToSync = @env.COMMON_API_BASE_URL@/id/countBenByVillageIdAndLastModifiedDate
```
Comment (drtechie) [inline src/main/environment/common_ci.properties:18]: Here we can use @env.HWC_IDENTITY_API_BASE@

So
@env.HWC_IDENTITY_API_BASE@ = base/hwc-identity-0.0.1/
and @env.COMMON_API_BASE_URL@ = base/hwc-commonapi-v1.0/

Comment (drtechie) [reply src/main/environment/common_ci.properties:18]: Will you please also edit 

getBenImageFromIdentity =@env.COMMON_API_BASE_URL@/id/benImageByBenRegID
to 
getBenImageFromIdentity =@env.HWC_IDENTITY_API_BASE@/id/benImageByBenRegID

File: src/main/java/com/iemr/hwc/service/choApp/CHOAppSyncServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+
+            ArrayList<Object[]> resList = benChiefComplaintRepo.getBenChiefComplaints(benRegID, visitCode);
+            ArrayList<BenChiefComplaint> benChiefComplaints = BenChiefComplaint.getBenChiefComplaints(resList);
+            response.put("BenChiefComplaints", benChiefComplaints);
+
+            //Fetching vitals
+            response.put("benAnthropometryDetail", benAnthropometryRepo.getBenAnthropometryDetail(benRegID, visitCode));
+            response.put("benPhysicalVitalDetail", benPhysicalVitalRepo.getBenPhysicalVitalDetail(benRegID, visitCode));
+
+            //Fetching History
```
Comment (drtechie) [inline src/main/java/com/iemr/hwc/service/choApp/CHOAppSyncServiceImpl.java:482]: Remove commented out code

**Comment by drtechie** [comment]
Please add copyright header to all files

File: src/main/environment/common_ci.properties (hunk 1 above)
Comment (Shrey475) [reply src/main/environment/common_ci.properties]: Done.

File: src/main/java/com/iemr/hwc/service/choApp/CHOAppSyncServiceImpl.java (hunk 2 above)
Comment (Shrey475) [reply src/main/java/com/iemr/hwc/service/choApp/CHOAppSyncServiceImpl.java]: Done.

**Comment by drtechie** [comment]
@devikasuresh20 / @ravishanigarapu  is it okay to merge this PR? Please check.

**Review by ravishanigarapu — APPROVED** [review]
looks ok

File: src/main/java/com/iemr/hwc/controller/choApp/CHOAppSyncController.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        this.choappSyncService = choappSyncService;
+    }
+
+    // beneficiary registration sync from CHO app to AMRIT server with common and identity new
+    @ApiOperation(value = "Sync new beneficiaries to AMRIT server", consumes = "application/json", produces = "application/json")
+    @RequestMapping(value = { "/beneficiariesToServer" }, method = { RequestMethod.POST })
+    public ResponseEntity<String> beneficiaryRegistrationSyncToServer(@RequestBody String comingReq,
+                                                              @RequestHeader(value = "Authorization") String Authorization) {
+
+        return choappSyncService.registerCHOAPPBeneficiary(comingReq, Authorization);
```
[the remark this answers is not in the document]
Comment (ravishanigarapu) [reply src/main/java/com/iemr/hwc/controller/choApp/CHOAppSyncController.java:58]: Please resolve all the CodeQL checks.

[the remark this answers is not in the document]
Comment (shreypatidar-beehyv) [reply src/main/java/com/iemr/hwc/controller/choApp/CHOAppSyncController.java:58]: Hi @ravishanigarapu, I think CodeQL has raised it as a reflected cross-site script vulnerability believing that the user-input HTTP header ("Authorization") is sent back in the response header as well, although that's not the case.
Also, not sure why warn for only these 3 controller methods, the method parameter
@RequestHeader(HttpHeaders.AUTHORIZATION) String Authorization
is used in the existing Registrar controller and many other controllers.
Maybe this is a false positive, Would appreciate your views on this.

**Comment by ravishanigarapu** [comment]
@drtechie  we not able to merge this PR.

**Comment by drtechie** [comment]
> This branch is out-of-date with the base branch

https://stackoverflow.com/questions/53712930/how-to-fix-this-in-git-this-branch-is-out-of-date-with-the-base-branch

**Comment by ravishanigarapu** [comment]
@shreypatidar-beehyv  please update this PR. 

https://stackoverflow.com/questions/53712930/how-to-fix-this-in-git-this-branch-is-out-of-date-with-the-base-branch

**Comment by shreypatidar-beehyv** [comment]
@drtechie , @ravishanigarapu There have been 2 new commits to 'develop' in the PSMRI HWC repo. One is an empty commit and another one has 832 file changes (the whole of the 'bin' folder has been pushed which has all the class files). Should I take these new updates into our forked repo as well? please confirm.

**Pull request merged**